### PR TITLE
Resnet 50 Retrain example enhancements

### PR DESCRIPTION
**Pull request by gigasquid**
I finally got time to look at the RESNET50 retraining example.

Thank you! It's great and surprisingly, it actually ran an locally on my mac!

I went through the documentation and enhanced it with things I ran into. I also added some functions to test a pic and label it. 

There is another function I added to continue the training epoch by epoch, but would appreciate feedback if it is correct.

Hope this helps - and keep up the great work 💯

### CODE REVIEW

File: examples/resnet-retrain/src/resnet_retrain/core.clj [hunk 1]
```diff
@@ -259,7 +259,42 @@
...
+(defn train-again
+  "incrementally improve upon the trained model"
   [& [batch-size]]
-  (train (when batch-size
-           (Integer/parseInt batch-size))))
+  (let [batch-size (or batch-size 32)
+        epoch-size 4096
+        network (util/read-nippy-file "trained-network.nippy")]
+    (println "training using batch size of" batch-size)
+    (train/train-n network
```
Comment (gigasquid) [inline examples/resnet-retrain/src/resnet_retrain/core.clj]: This is what I'm not sure of - is it doing what I think it should be doing here, taking the RESNET50 network with frozen layers and just retraining the last layer?

File: examples/resnet-retrain/resize_images.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 FILES=$(find data -name "*.tif")
 for x in $FILES; do
-    convert $x -resize 224x224 "${x/\.tif/\.png}"
+    convert $x -resize 224x224 "${x/\.tif/.png}"
```
Comment (gigasquid) [inline examples/resnet-retrain/resize_images.sh]: This I had to change on my Mac but unsure if I'm going to break it for other OS

File: examples/resnet-retrain/README.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # resnet-retrain
 
+## Snapshot version of Cortex
```
Comment (gigasquid) [inline examples/resnet-retrain/README.md]: I'll change this to non-snaphsot https://github.com/thinktopic/cortex/issues/237 to be in sync with https://github.com/thinktopic/cortex/issues/237

File: examples/resnet-retrain/resize_images.sh (hunk 2 above)
Comment (harold) [reply examples/resnet-retrain/resize_images.sh]: Yeah, this looks suspect. Why would the one dot need to be escaped but not the other?

File: examples/resnet-retrain/src/resnet_retrain/core.clj (hunk 1 above)
Comment (harold) [reply examples/resnet-retrain/src/resnet_retrain/core.clj]: I am not super-familiar with this path, but it sounds like the answer to your question depends on the contents of "trained-network.nippy". You could inspect the layers in there and see if the ones you expect to be frozen, are (that is, they have `:learning-attenuation` zero).

hth.

File: examples/resnet-retrain/resize_images.sh (hunk 2 above)
Comment (gigasquid) [reply examples/resnet-retrain/resize_images.sh]: not sure but my files generated ended up being renamed with a slash in it - I'll take it out of the commit. The heads up of what it should look like in the README should suffice to make sure people have the right filename

File: examples/resnet-retrain/src/resnet_retrain/core.clj (hunk 1 above)
Comment (gigasquid) [reply examples/resnet-retrain/src/resnet_retrain/core.clj]: thanks - I'll check it out.

File: examples/resnet-retrain/resize_images.sh (hunk 2 above)
Comment (harold) [reply examples/resnet-retrain/resize_images.sh]: Cool, yeah, I haven't tried this, so I can't say for sure. (:

File: examples/resnet-retrain/src/resnet_retrain/core.clj (hunk 1 above)
Comment (harold) [reply examples/resnet-retrain/src/resnet_retrain/core.clj]: Nice! Thank you too. This is an interesting area that can definitely use some attention so I'm glad you're here.

**Comment by cnuernber** [comment]
@gigasquid :  Let me know if I should land this PR or if you want to work on it some more.  I think you are ready but I am not certain.

**Comment by gigasquid** [comment]
@cnuernber I'd like a little more time. I'm going to make the code changes we talked about and I'm digging into verifying the layers of the resnet50 . I'll ping you when it is ready for another look. Thanks 😸

File: examples/resnet-retrain/src/resnet_retrain/core.clj (hunk 1 above)
Comment (gigasquid) [reply examples/resnet-retrain/src/resnet_retrain/core.clj]: yup - it is. The layers are all marked as `:non-trainable?`. The only ones that are trainable are the ones that are expected 
`({:node :labels, :non-trainable? nil} {:node :softmax-loss-1, :non-trainable? nil} {:node :fc21, :non-trainable? nil})`

Comment (harold) [reply examples/resnet-retrain/src/resnet_retrain/core.clj]: Hey, that's neat! Score one for models being just-clojure-data.

**Comment by gigasquid** [comment]
@cnuernber @harold I updated the PR to not use snapshots and reverted the change in the resize_images.sh. It's ready to go after you review.

**Pull request merged**